mq/esworker_guardduty.py

```diff
@@ -0,0 +1,170 @@
+#!/usr/bin/env python
+
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+# Copyright (c) 2017 Mozilla Corporation
+
+
+import json
+
+import sys
+import os
+import socket
+from configlib import getConfig, OptionParser
+from datetime import datetime
+from mozdef_util.utilities.toUTC import toUTC
+from mozdef_util.utilities.logger import logger, initLogger
+
+sys.path.insert(0, os.path.join(os.path.dirname(__file__), "../"))
+from esworker_sns_sqs import taskConsumer
+from mq.lib.plugins import sendEventToPlugins
+from mq.lib.sqs import connect_sqs
+from mozdef_util.elasticsearch_client import ElasticsearchClient
+
+
+# running under uwsgi?
+try:
+    import uwsgi
+
+    hasUWSGI = True
+except ImportError as e:
+    hasUWSGI = False
+
+
+class GDtaskConsumer(taskConsumer):
+    def build_submit_message(self, message):
+        # default elastic search metadata for an event
+        metadata = {"index": "events", "id": None}
+
+        event = {}
+
+        event["receivedtimestamp"] = toUTC(datetime.now()).isoformat()
+        event["mozdefhostname"] = self.options.mozdefhostname
+
+        if "tags" in event:
+            event["tags"].extend([self.options.taskexchange])
+        else:
+            event["tags"] = [self.options.taskexchange]
+
+        event["severity"] = "INFO"
+        event["source"] = "guardduty"
+        event["details"] = {}
+
+        event["details"] = message["details"]
+        if "hostname" in message:
+            event["hostname"] = message["hostname"]
+        if "summary" in message:
+            event["summary"] = message["summary"]
+        if "category" in message:
+            event["details"]["category"] = message["category"]
+        if "tags" in message:
+            event["details"]["tags"] = message["tags"]
+        event["utctimestamp"] = toUTC(message["timestamp"]).isoformat()
+        event["timestamp"] = event["utctimestamp"]
+        (event, metadata) = sendEventToPlugins(event, metadata, self.pluginList)
+        # Drop message if plugins set to None
+        if event is None:
+            return
+
+        self.save_event(event, metadata)
+
+    def on_message(self, message_raw):
+        if "Message" in message_raw:
+            message = json.loads(message_raw["Message"])
+            if "details" in message:
+                if "finding" in message["details"]:
+                    if "action" in message["details"]["finding"]:
+                        if "actionType" in message["details"]["finding"]["action"]:
```

Comment:
We can use the `key_exists` function from https://github.com/mozilla/MozDef/blob/master/mozdef_util/mozdef_util/utilities/key_exists.py to clean up these if statements, it'll change it to:

`if key_exists('details.finding.action.actionType', message):`